gems/sorbet-runtime/lib/types/props/unknown_value.rb

```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+# typed: strict
+
+module T::Props
+  class UnknownValue
```

Comment:
What name should this be called? `UnknownValue` makes it sound like you think this will only be used for enum serde failures, but the way you've implemented this, it works for literally any serde failure. That includes: value had the wrong type, key is missing, etc.

I think the name should mention something to that effect, like `DeserializationFailure` or something?

Reply:
> it works for literally any serde failure. That includes: value had the wrong type, key is missing, etc.

Yeah, that wasn't really the intention when I set off, just a bit of a nice outcome that it seems to possibly be more widely applicable.

But we don't really do type checks in deserialization as it is. It's probably a whole performance draining can of worms to do so, but maybe something we could start to allow opting into for new models. The one check we do perform is a nil-check, but that actually happens another layer up. We could move that downward to allow an unexpected nil to result in an `UnknownValue` as well.